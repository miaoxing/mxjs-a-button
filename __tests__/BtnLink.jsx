import {BtnLink} from '../';
import {MemoryRouter} from 'react-router';
import {render} from '@testing-library/react';
import {ConfigProvider} from 'antd';

describe('BtnLink', () => {
  test('renders correctly', () => {
    const {container} = render(
      <ConfigProvider theme={{hashed: false}}>
        <MemoryRouter>
          <BtnLink to="foo">bar</BtnLink>
        </MemoryRouter>
      </ConfigProvider>
    );

    expect(container).toMatchSnapshot();
  });

  test('text', () => {
    const {getByTestId} = render(
      <ConfigProvider theme={{hashed: false}}>
        <MemoryRouter>
          <BtnLink data-testid="foo" to="foo">bar</BtnLink>
        </MemoryRouter>
      </ConfigProvider>
    );

    expect(getByTestId('foo')).toMatchSnapshot();
  });
});
