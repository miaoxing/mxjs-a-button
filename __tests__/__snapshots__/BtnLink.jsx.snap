// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnLink renders correctly 1`] = `
<div>
  <a
    class="ant-btn ant-btn-default"
    href="/"
    tabindex="0"
  >
    <span>
      bar
    </span>
  </a>
</div>
`;

exports[`BtnLink text 1`] = `
<a
  class="ant-btn ant-btn-default"
  data-testid="foo"
  href="/foo"
  tabindex="0"
>
  <span>
    bar
  </span>
</a>
`;
